MatrixSDK/Data/Store/MXFileStore/MXFileStore.m

```diff
@@ -2279,4 +2256,63 @@ - (void)loadRoomMessagesForRoom:(NSString *)roomId completion:(void (^)(void))co
     });
 }
 
+#pragma mark - MXRoomSummaryStore
+
+- (NSArray<NSString *> *)rooms
+{
+    NSMutableArray<NSString *> *roomIDs = [NSMutableArray arrayWithArray:[[NSFileManager defaultManager] contentsOfDirectoryAtPath:storeRoomsPath error:nil]];
+    [roomIDs removeObjectsInArray:roomsToCommitForDeletion];
+    return roomIDs;
+}
+
+- (void)storeSummaryForRoom:(NSString *)roomId summary:(id<MXRoomSummaryProtocol>)summary
+{
+    @synchronized (roomSummaries)
```

Comment:
Are we ok that those methods will be removed from `MXFileStore`?
MXStore APIs are not thread safe and should be called only from the main thread. Having `@synchronized` is new for summary management.

Reply:
Discussed in IRL.
MXRoomSummaryStore should be separated from MXStore.
MXStore should not depend on MXRoomSummaryStore but MXMemoryStore or MXFileStore can implement MXRoomSummaryStore.